engines/dragons/actor.h

```diff
@@ -0,0 +1,161 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+#ifndef DRAGONS_ACTOR_H
+#define DRAGONS_ACTOR_H
+
+#include "common/system.h"
+
+namespace Dragons {
+class Actor;
+class ActorResourceLoader;
+class ActorResource;
+struct ActorFrame;
+
+#define DRAGONS_ENGINE_NUM_ACTORS 64
+
+enum ActorFlags {
+	ACTOR_FLAG_1 = 1,
+	ACTOR_FLAG_2 = 2,
+	ACTOR_FLAG_4 = 4,
+	ACTOR_FLAG_8 = 8,
+	ACTOR_FLAG_10 = 0x10, //actor is walking a path
+	ACTOR_FLAG_20 = 0x20,
+	ACTOR_FLAG_40 = 0x40,
+	ACTOR_FLAG_80 = 0x80,
+	ACTOR_FLAG_100 = 0x100,
+	ACTOR_FLAG_200 = 0x200,  // Use screen coordinates not map coordinates.
+	ACTOR_FLAG_400 = 0x400,
+	ACTOR_FLAG_800 = 0x800,
+	ACTOR_FLAG_1000 = 0x1000,
+	ACTOR_FLAG_2000 = 0x2000,
+	ACTOR_FLAG_4000 = 0x4000,
+	ACTOR_FLAG_8000 = 0x8000  //Seems turn off semi trans mode when selected.
+};
+
+enum ActorFrameFlags {
+	ACTOR_FRAME_FLAG_2 = 0x2,
+	ACTOR_FRAME_FLAG_10 = 0x10,
+	ACTOR_FRAME_FLAG_20 = 0x20
+};
+
+class ActorManager {
+public:
+	typedef Common::Array<Actor> Actors;
+	typedef Actors::iterator ActorsIterator;
+
+private:
+	ActorResourceLoader *_actorResourceLoader;
+	Actors _actors;
+	uint16 _displayOrder[DRAGONS_ENGINE_NUM_ACTORS];
+public:
+	ActorManager(ActorResourceLoader *actorResourceLoader);
+
+public:
+	Actor *loadActor(uint32 resourceId, uint32 sequenceId, int16 x, int16 y);
+	Actor *loadActor(uint32 resourceId, uint32 sequenceId, int16 x, int16 y, uint16 priorityLayer);
+	Actor *loadActor(uint32 resourceId, uint16 actorId);
+	Actor *getActor(uint16 actorId);
+	Actor *getActorByDisplayOrder(uint16 position);
+	void clearActorFlags(uint16 startingActorId);
+	ActorResource *getActorResource(uint32 resourceId);
+	void updateActorDisplayOrder();
+private:
+	Actor *findFreeActor(int16 resourceID);
+	void resetDisplayOrder();
+};
+
+class Actor {
+public:
+	uint16 _actorID;
+	ActorResource* _actorResource;
+	uint16 actorFileDictionaryIndex;
+	int16 resourceID;
+	byte *_seqCodeIp;
+	void* frame_pointer_maybe;
```

Comment:
Code formatting is wrong. Should be `void *frame_pointer_maybe;`, just as the line above.